### PR TITLE
User will now be able to see pdf files and they can change the pdf files from within the subtopic #3

### LINKED ISSUE #3: As a user, I can view each subtopic pdf
I should be able to switch subtopics by using the subtopc dropdown and the prev/next buttons.

Use react-pdf.js library to render the pdf
https://www.npmjs.com/package/react-pdf

Image

### PR DESCRIPTION
**Pull request by Soumik29**
1. User will be able to see the pdf Files
2. Did modification to the UI for the subject Page
3. Did some tweaking to change the size of the input of the search bar
4. Changed some CSS to make the background standout a bit
5. Made all the changes that were requested

### CODE REVIEW

File: src/pages/Subject.tsx [hunk 1]
```diff
@@ -1,54 +1,75 @@
...
-        to={ctx.getPreviousPath()}>
-        Back
-      </Button>
       <Typography variant="h3">Discrete Math</Typography>
       <Typography variant="subtitle1">
-        Category: {ctx.category?.name}
+        Category:{' '}
+        <Breadcrumbs aria-label="breadcrumb">
+          <NavLink to="/">Home</NavLink>
+          <NavLink to="http://localhost:5173/category/0/subject/0">
```
Comment (wztech0192) [inline src/pages/Subject.tsx]: we should not hard code this path, the category id/index should be getting from the context. And always start with `/`. In production, navigate to localhost:5173 will break the app

File: src/pages/Home.tsx [hunk 2]
```diff
@@ -58,9 +56,13 @@ export const Home = () => {
                               logical reasoning and proof techniques to solve
                               problems efficiently.
                             </Typography>
-                            <Button variant="outlined" size="medium">
+                            <Typography
+                              variant="button"
+                              color="primary"
+                              style={{ marginTop: '10px', display: 'block' }}
```
Comment (wztech0192) [inline src/pages/Home.tsx]: use `sx` instead of style

File: src/pages/Subject.tsx [hunk 3]
```diff
@@ -1,54 +1,75 @@
...
-                <Card variant="outlined" style={{height: '100%'}}>
+              <Grid item md={6} xs={12} key={si}>
+                <CustomStyledCard>
                   <CardContent>
-                    <Typography variant='h6' gutterBottom>
-                        {t.name}
-                    </Typography>
-                    <ul style={{listStyleType: "none"}}>
+                    <Typography variant="h5">{t.name}</Typography>
+                    <ul style={{ listStyleType: 'none' }}>
```
Comment (wztech0192) [inline src/pages/Subject.tsx]: use MUI's list component instead of native list

File: src/hooks/useRouteMetadataContext.ts [hunk 4]
```diff
@@ -9,13 +9,13 @@ export class RouteMetadataContext {
...
+//   // current category
   readonly category: Category | null;
-  // current subject
+//   // current subject
   readonly subject: Subject | null;
-  // current topic
+//   // current topic
   readonly topic: Topic | null;
-  // The hierarchy of topics leading to the current topic. The last element is the current topic.
+//   // The hierarchy of topics leading to the current t.opic. The last element is the current topic
```
Comment (wztech0192) [inline src/hooks/useRouteMetadataContext.ts]: remove additional `//`

**Review by wztech0192 — CHANGES_REQUESTED** [review]

File: src/pages/Subject.tsx (hunk 1 above)
Comment (Soumik29) [reply src/pages/Subject.tsx]: Changed the hard coded path and made it ctx.getSubjectPath()

File: src/pages/Home.tsx (hunk 2 above)
Comment (Soumik29) [reply src/pages/Home.tsx]: Removed style attribute and changed it to sx and applied the styling.

File: src/pages/Subject.tsx (hunk 3 above)
Comment (Soumik29) [reply src/pages/Subject.tsx]: Removed native list and used MUI's list component.

File: src/hooks/useRouteMetadataContext.ts (hunk 4 above)
Comment (Soumik29) [reply src/hooks/useRouteMetadataContext.ts]: Removed additional double slahses for the comments.

**Review by wztech0192 — APPROVED** [review]

**Pull request merged**